From f08e58fa89b6ef744eb69ae0d9aa908f25b13257 Mon Sep 17 00:00:00 2001
From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Tue, 11 Mar 2014 17:29:25 -0700
Subject: [PATCH 31/33] coverity scans: fix Unrecoverable parse warning

These errors cause distractions, even though they are low priority.
---
 src/thd_engine_default.cpp | 26 +++++++++++++++++---------
 1 file changed, 17 insertions(+), 9 deletions(-)

diff --git a/src/thd_engine_default.cpp b/src/thd_engine_default.cpp
index b75d899..f1158ef 100644
--- a/src/thd_engine_default.cpp
+++ b/src/thd_engine_default.cpp
@@ -37,15 +37,23 @@
 
 // Default CPU cooling devices, which are not part of thermal sysfs
 // Since non trivial initialization is not supported, we init all fields even if they are not needed
-static cooling_dev_t cpu_def_cooling_devices[] = {
-	{	.status = true,
-		.mask = CDEV_DEF_BIT_UNIT_VAL | CDEV_DEF_BIT_READ_BACK | CDEV_DEF_BIT_MIN_STATE | CDEV_DEF_BIT_STEP,
-		.index = 0, .unit_val = ABSOULUTE_VALUE, .min_state = 0, .max_state = 0, .inc_dec_step = 5,
-		.read_back = false, .auto_down_control = false,
-		.type_string = "intel_powerclamp", .path_str = "",
-		.debounce_interval = 4, .pid_enable = false,
-		.pid = {0.0, 0.0, 0.0}},
-};
+/* Some security scan handler can't parse, the following block and generate unnecessary errors.
+ * hiding good ones. So init in old style compatible to C++
+
+ static cooling_dev_t cpu_def_cooling_devices[] = {
+ {	.status = true,
+ .mask = CDEV_DEF_BIT_UNIT_VAL | CDEV_DEF_BIT_READ_BACK | CDEV_DEF_BIT_MIN_STATE | CDEV_DEF_BIT_STEP,
+ .index = 0, .unit_val = ABSOULUTE_VALUE, .min_state = 0, .max_state = 0, .inc_dec_step = 5,
+ .read_back = false, .auto_down_control = false,
+ .type_string = "intel_powerclamp", .path_str = "",
+ .debounce_interval = 4, .pid_enable = false,
+ .pid = {0.0, 0.0, 0.0}},
+ };
+ */
+static cooling_dev_t cpu_def_cooling_devices[] = { { true, CDEV_DEF_BIT_UNIT_VAL
+		| CDEV_DEF_BIT_READ_BACK | CDEV_DEF_BIT_MIN_STATE | CDEV_DEF_BIT_STEP,
+		0, ABSOULUTE_VALUE, 0, 0, 5, false, false, "intel_powerclamp", "", 4,
+		false, { 0.0, 0.0, 0.0 } } };
 
 cthd_engine_default::~cthd_engine_default() {
 	if (parser_init_done)
-- 
1.9.0

